### PR TITLE
initial version of the breadcrumb component

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

**Comment by ssilvert** [comment]
@edewit Please resolve conflict.

File: src/route-config.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  },
+  {
+    path: "/realm-roles",
+    component: RealmRolesSection,
+    breadcrumb: t("roles:roleList"),
+  },
+  {
+    path: "/users",
+    component: UsersSection,
+    breadcrumb: "Users",
```
Comment (ssilvert) [inline src/route-config.ts:64]: Please go ahead and localize these.

**Review by ssilvert — CHANGES_REQUESTED** [review]
@edewit This looks good.  Are there any screens where I can see it in action?

**Comment by srambach** [comment]
Making a component `Breadcrumbs` (plural) seems to ask for mistakes since there's already the PatternFly component `Breadcrumb` - could it be something like `PageBreadcrumb` instead?

File: package.json [hunk 2]
```diff
@@ -27,7 +27,8 @@
     "react-dom": "^16.8.5",
     "react-hook-form": "^6.8.2",
     "react-i18next": "^11.7.0",
-    "react-router-dom": "^5.2.0"
+    "react-router-dom": "^5.2.0",
+    "use-react-router-breadcrumbs": "^1.0.4"
```
Comment (dlabaj) [inline package.json:31]: We are adding a new library?  Do we really need this?

File: src/components/bread-crumb/__tests__/__snapshots__/BreadCrumbs.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`BreadCrumbs tests couple of crumbs 1`] = `
+<BreadCrumbs>
```
Comment (dlabaj) [inline src/components/bread-crumb/__tests__/__snapshots__/BreadCrumbs.test.tsx.snap:4]: Let's call this something else it's too close to Breadcrumb and can cause coding errors.

**Review by dlabaj — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (edewit) [reply package.json:31]: we don't we can copy the code this is all of the code: https://github.com/icd2k3/use-react-router-breadcrumbs/blob/master/src/index.tsx

**Comment by edewit** [comment]
@srambach I made it `PageBreadcrumbs` because it renders contains multiple crumbs IMO

File: package.json (hunk 2 above)
Comment (ssilvert) [reply package.json:31]: @dlabaj @edewit I'm not crazy about adding more libraries either.  But I think copying the code is even worse.  Let's just use the library.  Then at least we don't have to maintain the code.

**Review by ssilvert — CHANGES_REQUESTED** [review]
I can see it in action now.  Looks good except that there needs to be spacing between the breadcrumbs and the page title.

**Comment by ssilvert** [comment]
@edewit Oh, and also a conflict.

File: package.json (hunk 2 above)
Comment (dlabaj) [reply package.json:31]: agreed I don't want to copy the code.  Was just wondering if we really needed it or not.  It's small so we can bring it in.

**Review by dlabaj — APPROVED** [review]

**Comment by edewit** [comment]
> I can see it in action now. Looks good except that there needs to be spacing between the breadcrumbs and the page title.

@srambach I guess just adding some padding would is not the way to go for this?

**Comment by srambach** [comment]
> > I can see it in action now. Looks good except that there needs to be spacing between the breadcrumbs and the page title.
> 
> @srambach I guess just adding some padding would is not the way to go for this?

Probably not ;-) Can the breadcrumb be passed to the Page? That seems to be the way it's meant to be built now. (I think there used to be a page section variant for breadcrumbs but I don't see it now) https://www.patternfly.org/v4/components/page/react-demos

**Review by ssilvert — APPROVED** [review]

**Pull request merged**